src/main/java/com/amazon/ion/system/IonBinaryWriterBuilder_1_1.java

```diff
@@ -0,0 +1,189 @@
+
+    // TODO add auto-flush (see IonBinaryWriterBuilder.withAutoFlushEnabled)
+    // TODO consider adding stream-copy optimization (see IonBinaryWriterBuilder withStreamCopyOptimized)
+    // TODO consider adding user-configurable length prefix preallocation (see _Private_IonManagedBinaryWriterBuilder.withPaddedLengthPreallocation)
+    // TODO consider allowing symbol/macro table block size to be configured separately (see _Private_IonManagedBinaryWriterBuilder.withSymbolsBlockSize)
+    // TODO add Ion 1.1-specific configuration
+
+    /**
+     * Gets the size of the blocks of memory the writer will allocate to hold encoded bytes between flushes.
+     *
+     * @return the block size currently configured.
+     *
+     * @see #setBlockSize(int)
+     * @see #withBlockSize(int)
+     */
+    int getBlockSize();
+
+    /**
+     * Sets the size of the blocks of memory the writer will allocate to hold encoded bytes between flushes.
+     *
+     * @param size the block size in bytes. If unset, the default block size of 32768 bytes will be used.
+     *
+     * @see #getBlockSize()
+     * @see #withBlockSize(int)
+     */
+    void setBlockSize(int size);
+
+    /**
+     * Declares the size of the blocks of memory the writer will allocate to hold encoded bytes between flushes.
+     *
+     * @param size the block size in bytes. If unset, the default block size of 32768 bytes will be used.
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     *
+     * @see #getBlockSize()
+     * @see #setBlockSize(int)
+     */
+    IonBinaryWriterBuilder_1_1 withBlockSize(int size);
+
+    // NOTE: Unlike in Ion 1.0, local symbol table append is always enabled in the Ion 1.1 writers.
+    // NOTE: Unlike in Ion 1.0, writing float 32 is always enabled in the Ion 1.1 writers.
+
+    /**
+     * Creates a mutable copy of this builder.
+     *
+     * @return a new builder with the same configuration as {@code this}.
+     */
+    IonBinaryWriterBuilder_1_1 copy();
+
+    /**
+     * Returns an immutable builder configured exactly like this one.
+     *
+     * @return this instance, if immutable;
+     * otherwise an immutable copy of this instance.
+     */
+    IonBinaryWriterBuilder_1_1 immutable();
+
+    /**
+     * Returns a mutable builder configured exactly like this one.
+     *
+     * @return this instance, if mutable;
+     * otherwise a mutable copy of this instance.
+     */
+    IonBinaryWriterBuilder_1_1 mutable();
+
+    /**
+     * Builds a new writer based on this builder's configuration
+     * properties.
+     *
+     * @param out the stream that will receive Ion data.
+     * Must not be null.
+     *
+     * @return a new {@link IonWriter} instance; not {@code null}.
+     */
+    IonWriter build(OutputStream out);
+
+    // TODO add a build() method that returns a 1.1-specific writer interface, allowing opt-in to new APIs.
```

Comment:
Just a thought—should we consider having different `build()` functions for different levels of writers as well? I.e. raw writer, managed writer, etc.?

Reply:
Yes, we should consider it, as long as we make it clear which one most users will want. Facets have been used for this purpose, but they're a hassle.